transcom/mymove
# B 17716 tooltip component update

## [B-17716 Tooltip Component](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A850205&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

Created custom component for a tooltip that displays text on hover. Takes in three props:
`position`: can be either `left`, `right`, `bottom`, or `top` but when not passed in it defaults to `top`
`icon`: this displays the `<FontAwesomeIcon />` with passed in icon parameter. If left blank, it defaults to `circle-question`
`text`: this is the text that will display when hovering over or clicking the icon
`color`: this changes the color of the tooltip icon - can be HTML color code (like `#FF0000` for red) - default color is `blue`

An example of the ToolTip being used with all the bells & whistles and overrides defaults:
`<ToolTip text="this is the text that shows on hover" position="right" icon="circle-exclamation" color="#E43131" />`

Or you can keep it short and simple like this - which will display it on top with a blue `circle-question` icon:
`<ToolTip text="just keep it simple" />

I left the size of the text box fixed to keep positioning consistent. This will allow for constant results when displaying the text boxes on hover.

I also implemented a `useRef` functionality that hides all of the tooltip boxes when clicked outside of the `<div>` containing the tooltip. Multiple tooltips can be displayed and I did not implement an `onMouseLeave` function in case the office user needs to copy something and figured it'd be nice to leave the tooltip text displayed because hovering over a small icon can be a little finicky.

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Find any spot in the code and implement `<HoverTooltip text="Test text or whatever" icon="circle-question" position="left" /> or whatever you wish to pass in for initial props
2. Start the server & client and log in as an office user and navigate to where you implemented the tool tip
3. Hover over the icon and confirm text box displays in correct position
4. Click outside of the tooltip and confirm text box disappears
